src/tests/payment/test_stripe_webhook.py

```diff
@@ -0,0 +1,290 @@
+LASTNAME = "Doe"
+EMAIL = "johndoe@example.fr"
+USERNAME = EMAIL
+EMAIL_HOST_USER = "support@ordopro.fr"
+
+STRIPE_WEBHOOK_SECRET = "whsec_testsecret"
+
+
+@pytest.fixture
+def user(db):
+    """Creates and yields a new user."""
+    user = User.objects.create(
+        username=USERNAME, first_name=FIRSTNAME, last_name=LASTNAME, email=EMAIL
+    )
+    user.set_password(PASSWORD)
+    user.save()
+    yield user
+    user.delete()
+
+
+@pytest.fixture
+def client():
+    return Client()
+
+
+@pytest.fixture
+def payload():
+    """Fixture to generate a test payload."""
+    return {
+        "type": "checkout.session.completed",
+        "data": {
+            "object": {
+                "id": "cs_test_a1SdT2v81sae3cW",
+                "object": "checkout.session",
+                "amount_subtotal": 990,
+                "amount_total": 990,
+                "billing_address_collection": "required",
+                "cancel_url": "http://127.0.0.1:8000/api/v1/subscriptions/cancel/",
+                "created": 1731848739,
+                "currency": "eur",
+                "custom_fields": [],
+                "customer": "cus_REbNQXKKFCRF2c",
+                "customer_creation": "always",
+                "customer_details": {
+                    "address": {
+                        "city": "Saint-Ouen-l'Aumône",
+                        "country": "FR",
+                        "line1": "1 Rue Pagnere",
+                        "line2": "null",
+                        "postal_code": "95310",
+                        "state": "null",
+                    },
+                    "email": "kadi@test.com",
+                    "name": "kadi",
+                },
+                "customer_email": "kadi@test.com",
+                "expires_at": 1731935139,
+                "metadata": {"product_name": "Essentiel", "user_id": "5"},
+                "mode": "subscription",
+                "payment_status": "paid",
+                "status": "complete",
+            },
+        },
+    }
+
+
+def generate_stripe_signature(payload, secret):
+    """
+    Simulates the Stripe signature for the test using the secret key.
+    """
+    payload_str = json.dumps(payload)
+    timestamp = str(int(time.time()))
+    sig_base = f"{timestamp}.{payload_str}"
+    signature = hmac.new(
+        secret.encode("utf-8"), sig_base.encode("utf-8"), hashlib.sha256
+    ).hexdigest()
+    return f"t={timestamp},v1={signature}"
+
+
+@pytest.mark.django_db
+class TestStripeWebhook:
+
+    @mock.patch("stripe.Webhook.construct_event")
+    def test_checkout_session_completed_with_valid_signature(
+        self, mock_construct_event, client, payload, user
+    ):
+        payload["data"]["object"]["metadata"]["user_id"] = str(user.id)
+
+        sig_header = generate_stripe_signature(payload, STRIPE_WEBHOOK_SECRET)
+        mock_construct_event.return_value = payload
+        response = client.post(
+            reverse_lazy("v1:stripe-webhook"),
```

Comment:
DRY: Like in the other tests e.g. nurse test_views.py store the url in a member to reuse it across your other tests and then add a test_endpoint()